src/cmap/auth/gssapi.ts

```diff
@@ -30,60 +51,161 @@ export class GSSAPI extends AuthProvider {
       mechanismProperties['gssapiServiceName'] ||
       'mongodb';
 
-    const MongoAuthProcess = Kerberos.processes.MongoAuthProcess;
+    function externalCommand(
+      command: object,
+      cb: (
+        err: Error | MongoError | undefined,
+        result: { result: { payload: string; conversationId: any } }
+      ) => void
+    ) {
+      return connection.command('$external.$cmd', command, cb);
+    }
+
+    initialize(
+      {
+        retries: 10,
+        host,
+        port,
+        username,
+        password,
+        serviceName: gssapiServiceName,
+        canonicalizeHostName:
+          typeof mechanismProperties.gssapiCanonicalizeHostName === 'boolean'
+            ? mechanismProperties.gssapiCanonicalizeHostName
+            : false
+      },
+      (err, context) => {
+        if (err) return callback(err);
 
-    const authProcess = new MongoAuthProcess(host, port, gssapiServiceName, mechanismProperties);
+        context!.client.step('', (err, payload) => {
+          if (err) return callback(err);
 
-    authProcess.init(username, password, err => {
-      if (err) return callback(err, false);
-      authProcess.transition('', (err, payload) => {
-        if (err) return callback(err, false);
+          externalCommand(
```

Comment:
nit: I have a mild preference for creating the doc and then calling the method, just because it seems to generally reduce extra introduced whitespace:
```
const saslStartCommand = {
  saslStart: 1,
  mechanism: 'GSSAPI',
  payload,
  authoAuthorize: 1
};
client.command(..., saslStartCommand, ...);
```

Maybe a middle ground here would be to make a helper for these:
```
conn.command('$external.$cmd', makeSaslStart(payload), ...);
```